File: src/main/java/com/cuupa/dms/database/user/UserRepository.java
package com.cuupa.dms.database.user;

import org.springframework.data.repository.CrudRepository;

public interface UserRepository extends CrudRepository<DbUser, Integer> {

    DbUser findByUsername(String username);

    DbUser findByUsernameAndPassword(String username, String password);

    DbUser save(DbUser user);

    boolean existsUserByUsername(String username);
}

File: src/main/java/com/cuupa/dms/database/user/UserRepository.kt
package com.cuupa.dms.database.user

import org.springframework.data.repository.CrudRepository

interface UserRepository : CrudRepository<DbUser?, Int?> {
    fun findByUsername(username: String): DbUser?
    fun findByUsernameAndAccesstoken(username: String, accessToken: String): DbUser?
    fun findByUsernameAndPassword(username: String, password: String): DbUser?
    fun save(user: DbUser): DbUser?
    fun existsUserByUsername(username: String): Boolean
}